applications/ThermalDEMApplication/custom_elements/thermal_spheric_particle.cpp

```diff
@@ -0,0 +1,1443 @@
+      GetRealContactModel().AdjustContact(r_process_info, this);
+    }
+    else {
+      mNeighborDistanceAdjusted   = mNeighborDistance;
+      mNeighborSeparationAdjusted = mNeighborSeparation;
+      mContactRadiusAdjusted      = mContactRadius;
+    }
+
+    KRATOS_CATCH("")
+  }
+
+  //------------------------------------------------------------------------------------------------------------
+  void ThermalSphericParticle::StoreBallToBallContactInfo(const ProcessInfo& r_process_info, SphericParticle::ParticleDataBuffer& data_buffer, SphericParticle* neighbor, double GlobalContactForce[3], bool sliding) {
+    KRATOS_TRY
+
+    if (!mStoreContactParam)
+      return;
+
+    // Increment number of contact particle neighbors
+    mNumberOfContactParticleNeighbor++;
+
+    // Local relavive velocity components (normal and tangential)
+    std::vector<double> LocalRelativeVelocity{ data_buffer.mLocalRelVel[2], sqrt(data_buffer.mLocalRelVel[0] * data_buffer.mLocalRelVel[0] + data_buffer.mLocalRelVel[1] * data_buffer.mLocalRelVel[1]) };
+
+    // Local contact force components (normal and sliding tangential)
+    double LocalContactForce[3] = { 0.0 };
+    GeometryFunctions::VectorGlobal2Local(data_buffer.mLocalCoordSystem, GlobalContactForce, LocalContactForce);
+    std::vector<double> LocalForce{ LocalContactForce[2] };
+
+    // Friction heat generation is not considered when particles are not sliding against each other, so tangent velocity is set to zero.
+    // ATTENTION: Becareful when using the tangent velocity in other context that is not friction heat generation, as it can be zero.
+    if (sliding)
+      LocalForce.push_back(sqrt(LocalContactForce[0] * LocalContactForce[0] + LocalContactForce[1] * LocalContactForce[1]));
+    else
+      LocalForce.push_back(0.0);
+
+    // Update contact parameters
+    ContactParams params;
+    params.updated_step   = r_process_info[TIME_STEPS];
+    params.local_velocity = LocalRelativeVelocity;
+    params.local_force    = LocalForce;
+
+    // Keep impact parameters if contact is not new
+    if (mContactParamsParticle.count(neighbor)) {
+      params.impact_time     = mContactParamsParticle[neighbor].impact_time;
+      params.impact_velocity = mContactParamsParticle[neighbor].impact_velocity;
+    }
+    // Set impact parameters for new contacts
+    else {
+      params.impact_time     = r_process_info[TIME];
+      params.impact_velocity = LocalRelativeVelocity;
+    }
+
+    // Add/Update parameters in map
+    mContactParamsParticle[neighbor] = params;
+
+    KRATOS_CATCH("")
+  }
+
+  //------------------------------------------------------------------------------------------------------------
+  void ThermalSphericParticle::StoreBallToRigidFaceContactInfo(const ProcessInfo& r_process_info, SphericParticle::ParticleDataBuffer& data_buffer, DEMWall* neighbor, double GlobalContactForce[3], bool sliding) {
+    KRATOS_TRY
+
```

Comment:
Call to the base class method? Or you really want to discard the base class inf? I don't really know.

Reply:
Now calling empty base method.